actions/run.go

```diff
@@ -0,0 +1,149 @@
+package actions
+
+import (
+	"fmt"
+	"github.com/evoila/infraTESTure/config"
+	"github.com/evoila/infraTESTure/infrastructure"
+	"github.com/evoila/infraTESTure/infrastructure/bosh"
+	"github.com/evoila/infraTESTure/parser"
+	"github.com/urfave/cli"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"plugin"
+	"runtime"
+	"strings"
+)
+
+func Run(c *cli.Context) error {
+	conf, err := config.LoadConfig(c.String("config"))
+
+	if err != nil {
+		logError(err, "")
+		return err
+	}
+
+	bosh.InitInfrastructureValues(conf)
+
+	var repoPath string
+
+	if repoPath = appendSlash(os.TempDir()); conf.Github.SavingLocation != "" {
+		repoPath = appendSlash(conf.Github.SavingLocation)
+	}
+
+	repoPath += conf.Github.RepoName
+
+	if c.Bool("override") {
+		cmd := exec.Command("bash", "-c", "rm -rf "+repoPath)
+		err = cmd.Run()
+
+		if err != nil {
+			logError(err, "")
+			return err
+		}
+	}
+
+	// Check if the repository is already cloned, and if so use this repository
+	if _, err := os.Stat(repoPath); os.IsNotExist(err) {
+		log.Printf("[INFO] Cloning repository from %v\n", conf.Github.TestRepo)
+		err = gitClone(conf.Github.TestRepo, repoPath, conf.Github.Tag)
+		if err != nil {
+			logError(err, "Could not clone repository")
+			return err
+		}
+	} else {
+		log.Printf("[INFO] Using existing repository %v\n", repoPath)
+	}
+
+	serviceDir := repoPath + "/" + conf.Service.Name
+
+	// If the --edit flag is set open the test file instead of running the tests
+	if c.Bool("edit") {
+		//TODO: change since this fails with changes in infra-tests
+		cmd := exec.Command("bash", "-c", "code . "+serviceDir)
+		err = cmd.Run()
+
+		if err != nil {
+			logError(err, "Could not open test file")
+			return err
+		}
+	} else {
+		// Build the given test repository as a go plugin
+		log.Printf("[INFO] Building go plugin from directory %v\n", serviceDir)
+		cmd := exec.Command("bash", "-c", "cd "+serviceDir+" && "+runtime.Version()+" build -buildmode=plugin")
```

Comment:
"runtime.Version()" doesn't work here since "go1.xx.xx build"  is not a recognized command

Reply:
Fixed.